mayadata-io/mayastor-e2e
# feature: Generic test pipeline ER1-49, ER1-46, ER1-47

- testlists - use python and yaml file to deliver
    Define test lists in a more sane manner
    In Jenkins read the testlist using a python scripts this
    allows us to order and group tests as required.
    For now use ordering based on previous recorded data,
    namely long running tests are scheduled first.
    When using parallel stages work is more likely to be
    distributed equitably across the queues.
- remove MOAC support routines
- default branches to develop (control + data plane repos)
- Tweak test tag to be more compact
- rename xxRev to xxCommitOrTag - semantically accurate now
- pretty print hash map support
- Make DestroyCluster more robust
    - verify requested cluster has been destroyed
    - retry if not 3 times
- remove --logs options for e2e-test.sh
    - no need for logs on successful test runs
- ensure cluster environment is consistent prior
    to running tests.
- handle cluster creation failure by returning
    from parallel stage routine - scales down
    parallel stages running
- acquire test environment from cluster and
    record in xray
- create artificats for calling job
    - tests.txt - list of tests to run (can be empty)
    - failed_tests.txt - list of tests that failed
    - passed_tests.txt - list of tests that passed
- add support function RunTestJob for use by jobs
        invoking the generic test job.
- create generic-system-test Jenkins job
- fully parameterised job for
    - building images
    - running system tests
- create weekday jenkins job
- update create-install-image.sh to handle v1.0.0 as well as develop
- update e2e-test.sh to timeout tests after 2 hours (go test timeout)
- add support python script to record clutser test environment
    for xray
- reduce iterations for primitive_msp_delete to 30 from 100
    to allow completion within 2 hours
- fixup install to work with v1.0.0
- remove send xray report from selfci Jenkinsfile
- add maximumVolsIO configuration to selfci config
- try 10 times to delete ( observer 3 times failed)
  add a delay between tries
  do not fail if cluster delete failed, but emit messages
  further remediation may be required to delete unused clusters
- submodule update recursively
- do not default branches to develop,
  "library" code should not default any setting
- "fork" pipelines/common/common.groovy
  to pipelines/common2/common.groovy modified functionality,
  this preserves functionality for existing pipeline scripts
  it is anticipated that future changes will refactor/remove
  those pipelines eventually and pipelines/common/common.groovy
  will not be needed.
- assert that formal parameters passed in as maps existing
  prior to using them
